commit b87891f5aff91b8ebbda8d9218009495848f7747
Author: Martin Cermak <mcermak@redhat.com>
Date:   Thu May 16 16:51:08 2024 +0200

    PR31495: teach stap-prep to work with other RT kernels

diff --git a/stap-prep b/stap-prep
index 8b429f880..2bbb6bc2f 100755
--- a/stap-prep
+++ b/stap-prep
@@ -103,13 +103,16 @@ done
 # 5.14.0-200.rt14.201.el9 ->
 #      "kernel-rt-debug-5.14.0-200.rt14.201.el9"
 # OR?! "kernel-rt-5.14.0-200.rt14.201.el9"
-if expr "$UNAME" : ".*\.rt.*" >/dev/null;
+# OR??!"kernel-rt-5.14.0-447.el9.x86_64+rt"
+if expr "$UNAME" : ".*\.rt.*" || expr "$UNAME" : ".*\+rt.*" >/dev/null;
 then
     KERNEL=`echo $KERNEL | sed -e s,kernel,kernel-rt,`
 fi
 
 KERN_ARCH=`uname -m`
-KERN_REV=`echo $UNAME | sed s/.$KERN_ARCH//` # strip arch from uname
+# strip arch from uname, for kernels like 5.14.0-447.el9.x86_64+rt or
+# 6.9.0-0.rc2.1.el10.x86_64+rt strip the +rt suffix too
+KERN_REV=`echo $UNAME | sed s/.$KERN_ARCH// | sed s/\+rt$//`
 if [ -x /usr/bin/dnf4 ]; then
     DI="dnf4 debuginfo-install"
     DI_DEPS=""
